MISSING REPLACEMENT DATA FOR RUN_DIR when using Gradle 4.5

1. Download the latest forge MDK.
3. Check `./gradlew -v` to see that the version is 2.14.
2. Run `./gradlew build` &rarr; everything works
4. Update the gradle wrapper to 4.4.1:

    ```./gradlew wrapper --gradle-version=4.4.1```

5. Run `./gradlew build` &rarr; everything works
6. Update the gradle wrapper to 4.5:

    ```./gradlew wrapper --gradle-version=4.5```

7. Run `./gradlew build` &rarr; failure

The build fails with the following message (`runDir` is still specified in build.gradle; build.gradle hasn't been edited at all during this process):

```
FAILURE: Build failed with an exception.

* Where:
Build file 'C:\Users\Pokechu22\Downloads\forge-1.12.2-14.23.1.2604-mdk\build.gradle' line: 10

* What went wrong:
A problem occurred evaluating root project 'forge-1.12.2-14.23.1.2604-mdk'.
> Failed to apply plugin [id 'net.minecraftforge.gradle.forge']
   > MISSING REPLACEMENT DATA FOR RUN_DIR

* Try:
Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output. Run with --scan to get full insights.

* Get more help at https://help.gradle.org

BUILD FAILED in 48s
```

Full log, with `--debug --stacktrace`: https://gist.github.com/Pokechu22/b91a8447d6977cb9a376b488c6622871

--- a/src/main/java/net/minecraftforge/gradle/user/UserBasePlugin.java
+++ b/src/main/java/net/minecraftforge/gradle/user/UserBasePlugin.java
@@ -902,7 +902,12 @@ public abstract class UserBasePlugin<T extends UserBaseExtension> extends BasePl
             JavaExec exec = makeTask("runClient", JavaExec.class);
             exec.getOutputs().dir(delayedFile(REPLACE_RUN_DIR));
             exec.setMain(GRADLE_START_CLIENT);
-            exec.workingDir(delayedFile(REPLACE_RUN_DIR));
+            exec.doFirst(new Action<Task>() {
+                @Override
+                public void execute(Task task) {
+                    ((JavaExec) task).workingDir(delayedFile(REPLACE_RUN_DIR));
+                }
+            });
             exec.setStandardOutput(System.out);
             exec.setErrorOutput(System.err);
 
@@ -919,7 +924,12 @@ public abstract class UserBasePlugin<T extends UserBaseExtension> extends BasePl
             JavaExec exec = makeTask("runServer", JavaExec.class);
             exec.getOutputs().dir(delayedFile(REPLACE_RUN_DIR));
             exec.setMain(GRADLE_START_SERVER);
-            exec.workingDir(delayedFile(REPLACE_RUN_DIR));
+            exec.doFirst(new Action<Task>() {
+                @Override
+                public void execute(Task task) {
+                    ((JavaExec) task).workingDir(delayedFile(REPLACE_RUN_DIR));
+                }
+            });
             exec.setStandardOutput(System.out);
             exec.setStandardInput(System.in);
             exec.setErrorOutput(System.err);

--- a/src/main/java/net/minecraftforge/gradle/user/UserBasePlugin.java
+++ b/src/main/java/net/minecraftforge/gradle/user/UserBasePlugin.java
@@ -902,12 +902,7 @@ public abstract class UserBasePlugin<T extends UserBaseExtension> extends BasePl
             JavaExec exec = makeTask("runClient", JavaExec.class);
             exec.getOutputs().dir(delayedFile(REPLACE_RUN_DIR));
             exec.setMain(GRADLE_START_CLIENT);
-            exec.doFirst(new Action<Task>() {
-                @Override
-                public void execute(Task task) {
-                    ((JavaExec) task).workingDir(delayedFile(REPLACE_RUN_DIR));
-                }
-            });
+            exec.doFirst(task -> ((JavaExec) task).workingDir(delayedFile(REPLACE_RUN_DIR)));
             exec.setStandardOutput(System.out);
             exec.setErrorOutput(System.err);
 
@@ -924,12 +919,7 @@ public abstract class UserBasePlugin<T extends UserBaseExtension> extends BasePl
             JavaExec exec = makeTask("runServer", JavaExec.class);
             exec.getOutputs().dir(delayedFile(REPLACE_RUN_DIR));
             exec.setMain(GRADLE_START_SERVER);
-            exec.doFirst(new Action<Task>() {
-                @Override
-                public void execute(Task task) {
-                    ((JavaExec) task).workingDir(delayedFile(REPLACE_RUN_DIR));
-                }
-            });
+            exec.doFirst(task -> ((JavaExec) task).workingDir(delayedFile(REPLACE_RUN_DIR)));
             exec.setStandardOutput(System.out);
             exec.setStandardInput(System.in);
             exec.setErrorOutput(System.err);

